### PR TITLE
If tor V2/V3 addr used in startup cmdline don't fail

### LINKED ISSUE #3090: Failed if onion addr is defined at cmdline
### Issue and Steps to Reproduce

```
lightningd --network=regtest   --proxy=127.0.0.1:9050 --addr=owtq3ffjulnk5m6f.onion   --bind-addr=10.0.2.15:1235 --addr=127.0.0.1:9735
lightning_connectd: Invalid listener wireaddress type 3 (version v0.7.2.1-291-g14b55d0)
0x557b2b1c30b2 send_backtrace
	common/daemon.c:40
0x557b2b1c5e0e status_failed
	common/status.c:199
0x557b2b1bc23d handle_wireaddr_listen
	connectd/connectd.c:906
0x557b2b1bc94a setup_listeners
	connectd/connectd.c:1091
0x557b2b1bcb4a connect_init
	connectd/connectd.c:1184
0x557b2b1be01d recv_req
	connectd/connectd.c:1476
0x557b2b1c3206 handle_read
	common/daemon_conn.c:31
0x557b2b1e0b4c next_plan
	ccan/ccan/io/io.c:59
0x557b2b1e1034 do_plan
	ccan/ccan/io/io.c:407
0x557b2b1e1061 io_ready
	ccan/ccan/io/io.c:417
0x557b2b1e2833 io_loop
	ccan/ccan/io/poll.c:445
0x557b2b1be6ab main
	connectd/connectd.c:1577
0x7f8c3c5612e0 ???
	???:0
0x557b2b1b9f79 ???
	???:0
0xffffffffffffffff ???
	???:0
2019-09-24T07:24:09.148Z **BROKEN** lightning_connectd(19502): Invalid listener wireaddress type 3 (version v0.7.2.1-291-g14b55d0)
2019-09-24T07:24:09.148Z **BROKEN** lightning_connectd(19502): backtrace: common/daemon.c:45 (send_backtrace) 0x557b2b1c30fa
2019-09-24T07:24:09.148Z **BROKEN** lightning_connectd(19502): backtrace: common/status.c:199 (status_failed) 0x557b2b1c5e0e
2019-09-24T07:24:09.148Z **BROKEN** lightning_connectd(19502): backtrace: connectd/connectd.c:906 (handle_wireaddr_listen) 0x557b2b1bc23d
2019-09-24T07:24:09.149Z **BROKEN** lightning_connectd(19502): backtrace: connectd/connectd.c:1091 (setup_listeners) 0x557b2b1bc94a
2019-09-24T07:24:09.149Z **BROKEN** lightning_connectd(19502): backtrace: connectd/connectd.c:1184 (connect_init) 0x557b2b1bcb4a
2019-09-24T07:24:09.149Z **BROKEN** lightning_connectd(19502): backtrace: connectd/connectd.c:1476 (recv_req) 0x557b2b1be01d
2019-09-24T07:24:09.149Z **BROKEN** lightning_connectd(19502): backtrace: common/daemon_conn.c:31 (handle_read) 0x557b2b1c3206
2019-09-24T07:24:09.150Z **BROKEN** lightning_connectd(19502): backtrace: ccan/ccan/io/io.c:59 (next_plan) 0x557b2b1e0b4c
2019-09-24T07:24:09.150Z **BROKEN** lightning_connectd(19502): backtrace: ccan/ccan/io/io.c:407 (do_plan) 0x557b2b1e1034
2019-09-24T07:24:09.150Z **BROKEN** lightning_connectd(19502): backtrace: ccan/ccan/io/io.c:417 (io_ready) 0x557b2b1e1061
2019-09-24T07:24:09.150Z **BROKEN** lightning_connectd(19502): backtrace: ccan/ccan/io/poll.c:445 (io_loop) 0x557b2b1e2833
2019-09-24T07:24:09.150Z **BROKEN** lightning_connectd(19502): backtrace: connectd/connectd.c:1577 (main) 0x557b2b1be6ab
2019-09-24T07:24:09.151Z **BROKEN** lightning_connectd(19502): backtrace: (null):0 ((null)) 0x7f8c3c5612e0
2019-09-24T07:24:09.151Z **BROKEN** lightning_connectd(19502): backtrace: (null):0 ((null)) 0x557b2b1b9f79
2019-09-24T07:24:09.151Z **BROKEN** lightning_connectd(19502): backtrace: (null):0 ((null)) 0xffffffffffffffff
2019-09-24T07:24:09.151Z **BROKEN** lightning_connectd(19502): STATUS_FAIL_INTERNAL_ERROR: Invalid listener wireaddress type 3
lightningd: lightning_connectd failed (exit status 242), exiting.
```

### `getinfo` output
<details><summary>show</summary>
```
{
   "id": "0279df3dedb790e1dce52cd2ded846b0f939be96f217c290062996b16d7b0d3492",
   "alias": "CHILLYGLEE",
   "color": "0279df",
   "num_peers": 1,
   "num_pending_channels": 0,
   "num_active_channels": 1,
   "num_inactive_channels": 0,
   "address": [
      {
         "type": "torv2",
         "address": "owtq3ffjulnk5m6f.onion",
         "port": 9735
      }
   ],
   "binding": [
      {
         "type": "ipv4",
         "address": "10.0.2.15",
         "port": 1235
      },
      {
         "type": "ipv4",
         "address": "127.0.0.1",
         "port": 9735
      }
   ],
   "version": "v0.7.2.1-291-g14b55d0",
   "blockheight": 1061,
   "network": "regtest",
   "msatoshi_fees_collected": 0,
   "fees_collected_msat": "0msat"
}
```
</details>

### PR DESCRIPTION
**Pull request by Saibato**
Fix  #3090

### CODE REVIEW

**Comment by Saibato** [comment]
Not direct related. After 3 forced push with exactly the same PR I won the travis lottery upstream
@rustyrussell do we really test the code or is this this an IQ test and a random green checker generator to show some care and to yield liabities ?  :1234:

**Comment by rustyrussell** [comment]
`--addr` means you want to listen on that address.  You can't listen on an onion address (it's not supported at least; we could use the tor control socket to try).  We should catch this in the front-end parser though: "I think you want --announce-addr".

This change would make `--bind-addr=torv2` "work" but it wouldn't do anything.

**Comment by Saibato** [comment]
> You can't listen on an onion address
Yes, the tor deamon does this for you in a proxy like maner, and all you can do is provide a socket  ( ip.port ) you listen and bind to and if you announce indicate to others that that onion:port maps to your node.
So to have a line --addr=vx.onion:[IP]:[PORT] would be much more useful, if we would auto create also the node ip socket the onion should map to and also would change on the fly the torrc file and then respawn tor, if we don't want service calls. 

> This change would make `--bind-addr=torv2` "work" but it wouldn't do anything.

PR is meant to be an uggly crash prevent in terms of UX since we use that full addr semantic with autotor  that also creates no new sockets when called with --addr=autortor,  onions are used  kind of .onion <-> IP pairs, in fact as you see it also, every address ( ip,onion, etc ...) is somehow a label mapping to a socket ( that could have strange underlying level transport even sneaker or rfc1149 :smile:  ) we use already with autotor the semantic --addr --bind-addr --announce-addr without crash fail, since we provide the bind under the hood to our default ip or a IP defined in an extra line with -add/bind-addr=IP.

So you say, and that is exactly so it does not much in terms of functionality.
I think with no crash the UX is more intuitive and at least those --addr=vx.onion lines  would show up in getinfo, what most secondary apps use as source of truth about the behavior of the node.  And if you say in cmd line --addr=torv3:port and in torrc you have this line torv3 port_in:local IP:port_out  your node does exactly this. The local addr must not be announce able and you could have a lot of local addr bind to loopback other sockets but your addr is for every other node only this onion addr ID pair your define with --addr or -bind-addr sure --bind-addr does really nothing
but some log entry and is just a cmdline reminder that you might have in torrc still a binding defined to your node you might forget about. like the bullet in a pistol, that is still in the barrel if you just drop  only your magazine.    

One thing i could imagine, is to extend the --addr or --bind option when called with predefined onion to create also a new socket, so your line could be --addr=vx.onion:[IP]:[PORT]  what in fact is the same as to type
--annouce-addr=vx.onion --bind-addr=[IP]:[PORT]  just a bit shorter. and the intended binding is more clear to the node operator.   
     
Since we have with normal tor the 'real' binding onion socket:port defined (if not with service call autotor) just in torrc, it's not clear from the  --announce-addr line or the getinfo call to what IP our onion binds or if it binds to your node at all even tough you can announce it.    

hmm ..?

File: lightningd/options.c [hunk 1]
```diff
@@ -193,6 +183,36 @@ static char *opt_add_announce_addr(const char *arg, struct lightningd *ld)
 	return NULL;
 }
 
+static char *opt_add_addr(const char *arg, struct lightningd *ld)
+{
+
+/* handle In case you used an .onion*/
+	if (strends(arg, ".onion")) {
+			printf("You used  ** --addr=%s ** option with an .onion address, please use"
```
[the remark this answers is not in the document]
Comment (Saibato) [reply lightningd/options.c]: Yes, that is unusual :smile: simsalabim tree times knocked,  change as you wished done...

File: lightningd/options.c [hunk 2]
```diff
@@ -193,6 +183,34 @@ static char *opt_add_announce_addr(const char *arg, struct lightningd *ld)
...
 }
 
+static char *opt_add_addr(const char *arg, struct lightningd *ld)
+{
+	/* handle in case you used the addr option with an .onion */
+	if (strends(arg, ".onion")) {
+			log_unusual(ld->log, "You used  ** --addr=%s ** option with an .onion address, please use"
+					" --announce-addr ! You are lucky in this node live some wizards and"
+					" fairies, we have done this for you and announce, Be as hidden as wished"
+					, arg);
```
Comment (cdecker) [inline lightningd/options.c]: That `,` belongs on the previous line.

File: lightningd/options.c [hunk 3]
```diff
@@ -193,6 +183,34 @@ static char *opt_add_announce_addr(const char *arg, struct lightningd *ld)
 	return NULL;
 }
 
+static char *opt_add_addr(const char *arg, struct lightningd *ld)
+{
+	/* handle in case you used the addr option with an .onion */
+	if (strends(arg, ".onion")) {
+			log_unusual(ld->log, "You used  ** --addr=%s ** option with an .onion address, please use"
```
Comment (cdecker) [inline lightningd/options.c]:
```suggestion
			log_unusual(ld->log, "You used `--addr=%s` option with an .onion address, please use"
```

File: lightningd/options.c [hunk 4]
```diff
@@ -193,6 +183,34 @@ static char *opt_add_announce_addr(const char *arg, struct lightningd *ld)
...
+					" --announce-addr ! You are lucky in this node live some wizards and"
```
Comment (cdecker) [inline lightningd/options.c]:
```suggestion
					" `--announce-addr`! You are lucky in this node live some wizards and"
```

File: lightningd/options.c [hunk 5]
```diff
@@ -193,6 +183,34 @@ static char *opt_add_announce_addr(const char *arg, struct lightningd *ld)
...
+	}
+	/* the intended call */
+	return opt_add_addr_withtype(arg, ld, ADDR_LISTEN_AND_ANNOUNCE, true);
+}
+
+static char *opt_add_bind_addr(const char *arg, struct lightningd *ld)
+{
+	/* handle in case you used the bind option with an .onion */
+	if (strends(arg, ".onion")) {
+			log_unusual(ld->log, "You used  ** --addr-bind=%s ** option with an .onion address,"
```
Comment (cdecker) [inline lightningd/options.c]:
```suggestion
			log_unusual(ld->log, "You used `--addr-bind=%s` option with an .onion address,"
```

File: lightningd/options.c (hunk 2 above)
Comment (Saibato) [reply lightningd/options.c]: done :+1:

File: lightningd/options.c (hunk 3 above)
Comment (Saibato) [reply lightningd/options.c]: done :+1:

File: lightningd/options.c (hunk 4 above)
Comment (Saibato) [reply lightningd/options.c]: done :+1:

File: lightningd/options.c (hunk 5 above)
Comment (Saibato) [reply lightningd/options.c]: done :+1: And changed `if (strends(arg, ".onion")) {` to a more  bolder check to allow xy.onion:[port]. 

siggh: The first commit with just `return` false; in connectd.c was not so long but
          @rustyrussell idea with "I think you want ..." resonated, so ...
and now I have found at other places also some new issues that need extra PR, please nack so I can go back in my mirror :smile: and tell you only what you wish for ...

**Comment by ZmnSCPxj** [comment]
ACK 832b2ac

**Comment by rustyrussell** [comment]
New message is nicer.  I would have failed them outright, but the message should scare them :)

**Pull request merged**